flow/connectors/mongo/mongo.go

```diff
@@ -405,3 +409,41 @@ func defaultPipeline() mongo.Pipeline {
 		}}},
 	}
 }
+
+func parseAsClientOptions(config *protos.MongoConfig) (*options.ClientOptions, error) {
+	connStr, err := connstring.Parse(config.Uri)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing uri: %w", err)
+	}
+
+	clientOptions := options.Client().
+		ApplyURI(config.Uri).
+		SetAppName("PeerDB Mongo Connector").
+		SetAuth(options.Credential{
+			Username: config.Username,
+			Password: config.Password,
+		})
+
+	if connStr.Compressors != nil {
+		clientOptions.SetCompressors([]string{"zstd", "snappy"})
+	}
+	if connStr.ReadPreference == "" {
+		clientOptions.SetReadPreference(readpref.SecondaryPreferred())
```

Comment:
do we want this? is this recommended somewhere?

Reply:
discussed offline, defaulting to secondary for CDC to avoid affecting production traffic -- but give user ability to override if they wish.

Reply:
discussed offline, defaulting to secondary for CDC to avoid affecting production traffic -- but give user ability to override if they wish.